android/app/src/main/java/poke/rogue/helper/presentation/dex/detail/PokemonDetailViewModel.kt

```diff
@@ -74,6 +80,31 @@ class PokemonDetailViewModel(
         }
     }
 
+    override fun navigateToBattle(battlePopUpUiModel: BattlePopUpUiModel) {
+        viewModelScope.launch {
+            val successState =
+                uiState
+                    .filterIsInstance<PokemonDetailUiState.Success>()
+                    .first()
+
+            _navigateToPokemonDetailToBattleEvent.emit(
+                PokemonDetailToBattleEvent(
+                    battlePopUp = battlePopUpUiModel,
+                    pokemon = successState.pokemon,
+                ),
+            )
+
+            launch {
+                analyticsLogger().logPokemonDetailToBattle(
```

Comment:
```suggestion
                logger.logPokemonDetailToBattle(
```

1) 이미 존재하는 logger를 써주세요~
2) launch 로 analytics 로직을 감싸는 것은 바람직하지 않습니다. 만약 아날릭틱스 로거에서 예외가 발생하면 `navigateToBattle` 는 예외를 던지게 됩니다.  

관련해서 좋은 블로그가 있어 추천드려요!

https://velog.io/@murjune/kotlin-Coroutine-supervisorScope-vs-SupervisorJob-%EC%96%B4%EB%96%A4%EA%B1%B8-%EC%82%AC%EC%9A%A9%ED%95%98%EB%9D%BC%EB%8A%94%EA%B1%B0%EC%A7%80#4-%EC%8B%A4-%EC%82%AC%EB%A1%80-analyticsscope